### ISSUE TITLE
Refine color system and unify alerts

### ISSUE THREAD

**Issue by connium** [digitalfabrik/entitlementcard#3132]
### Describe the Problem

In #2744 I mentioned, that the alerts don't have a sufficient contrast.
While working on it, @f1sh1918 [figured out](https://github.com/digitalfabrik/entitlementcard/issues/2744#issuecomment-5506514568), that we override the MUI defaults with [our own color system](https://www.figma.com/design/SDoGnXIjCCXpE1tAAJkRVc/entitlementcard?node-id=2990-690&p=f&t=CP3myTeHjeKXPsQx-0).

Additionally, @ulyanahoy-cloud mentioned, that we should use the `default` style for alerts instead of the `outlined` one.

### Describe Your Preferred Solution

- Revise the color system, to meet the a11y criteria (or use the MUI defaults for `error`, `warning`, `success`, `info` states).
- Adapt the code to the revised color system.
- Apply `default` style to all alerts.

### Describe Possible Alternatives

N/A

**Comment by f1sh1918** [comment]
Here are some suggestions i've tested

Good idea, since some severities like success and warning are okay but others not readable we should ensure to have the same contrast ratio for all light colors within our color system. If we can guarantee 2.00, all texts should work properly
Since MUI has lighten or darken functions depending on the use case of the color, we don't have to define the light colors with actual target contrast.


  | Farbe | `light` (aktuell) | Kontrast (aktuell) | `light` (Vorschlag, ~2.00:1) | Kontrast (neu) |
  |---|---|---:|---|---:|
  | primary | `#99A7D9` | 2.36:1 | `#abb6e0` | 2.00:1 |
  | error | `#ffcbce` | 1.43:1 | `#ff9ba1` | 2.01:1 |
  | warning | `#ffb64e` | 1.74:1 | `#ffa21f` | 2.01:1 |
  | success | `#81C784` | 2.01:1 | `#82c785` | 2.01:1 |
  | info | `#d7edf5` | 1.21:1 | `#76c1dd` | 2.01:1 |

Example (info color used for text) @connium and @ulyanahoy-cloud 

Before:

<img width="254" height="395" alt="Image" src="https://github.com/user-attachments/assets/fbcd79b2-b402-45fc-acce-73597cb8a747" />

After:
<img width="256" height="434" alt="Image" src="https://github.com/user-attachments/assets/8913b357-5016-4e2e-9509-eed98955e224" />

**Comment by ulyanahoy-cloud** [comment]
Since MUI can lighten or darken colors depending on the use case, I wouldn’t necessarily use 2:1 as a fixed accessibility target for every .light color.  What matters most is that the final color combinations in the components meet the required minimum contrast of 4.5:1 for text and 3:1 for icons.

For Alerts, I would still prefer using the standard MUI variant consistently.

**Comment by f1sh1918** [comment]
I understand but even we don't use outline variant, with current light color the background of the standard alert is way too light for these colors. Even we don't need 2:1 it should be at least equal for all colors IMO
Currently error and info is way too light

**Comment by ulyanahoy-cloud** [comment]
Yes, that makes sense. I agree with you.

**Comment by connium** [comment]
I've calculated the color contrasts for all our system colors. Details can be found in [Nextcloud](https://nextcloud.tuerantuer.org/index.php/f/6787087).

For some colors, neither `#ffffff` (surface light) nor `#25262e` (surface dark) give us sufficient contrast.
For many colors, the used foreground color in [color scheme](https://www.figma.com/design/SDoGnXIjCCXpE1tAAJkRVc/entitlementcard?node-id=2990-690&p=f&t=obFOVSzhtuZ742HA-0) is not the appropriate one.
